src/TensorNetwork.jl

```diff
@@ -3,109 +3,57 @@ using Random
 using EinExprs
 using OMEinsum
 using ValSplit
+using Classes
 
 """
-    Ansatz
-
-Type representing the predefined form of the Tensor Network.
-"""
-abstract type Ansatz end
-
-"""
-    Arbitrary
-
-Tensor Networks without a predefined form.
-"""
-abstract type Arbitrary <: Ansatz end
-
-# NOTE currently, these are implementation details
-function checkmeta end
-function metadata end
-
-"""
-    TensorNetwork{Ansatz}
+    TensorNetwork
 
 Graph of interconnected tensors, representing a multilinear equation.
 Graph vertices represent tensors and graph edges, tensor indices.
 """
-struct TensorNetwork{A<:Ansatz,M<:NamedTuple}
+@class TensorNetwork begin
     indices::Dict{Symbol,Vector{Int}}
     tensors::Vector{Tensor}
-    metadata::M
-
-    function TensorNetwork{A}(tensors; metadata...) where {A}
-        indices = reduce(enumerate(tensors); init = Dict{Symbol,Vector{Int}}([])) do dict, (i, tensor)
-            mergewith(vcat, dict, Dict([index => [i] for index in inds(tensor)]))
-        end
-
-        # Check for inconsistent dimensions
-        for (index, idxs) in indices
-            allequal(Iterators.map(i -> size(tensors[i], index), idxs)) ||
-                throw(DimensionMismatch("Different sizes specified for index $index"))
-        end
-
-        M = Tenet.metadata(A)
-        metadata = M((; metadata...))
-
-        tn = new{A,M}(indices, tensors, metadata)
-
-        checkansatz(tn)
-        return tn
-    end
 end
 
-TensorNetwork{A}(; metadata...) where {A<:Ansatz} = TensorNetwork{A}(Tensor[]; metadata...)
-
-# ansatz defaults to `Arbitrary`
-TensorNetwork(args...; kwargs...) = TensorNetwork{Arbitrary}(args...; kwargs...)
-
-# TODO maybe rename it as `convert` method?
-TensorNetwork{A}(tn::TensorNetwork{B}; metadata...) where {A,B} =
-    TensorNetwork{A}(tensors(tn); merge(tn.metadata, metadata)...)
-
-# TODO do sth to skip checkansatz? like @inbounds
-function checkansatz(tn::TensorNetwork{A}) where {A<:Ansatz}
-    for T in superansatzes(A)
-        checkmeta(T, tn) || throw(ErrorException("\"$T\" metadata is not valid"))
+TensorNetwork() = TensorNetwork(Tensor[])
+function TensorNetwork(tensors)
```

Comment:
```suggestion
TensorNetwork(tensors::Vector{<:Tensor}) = TensorNetwork(Tensor[tensors...])

function TensorNetwork(tensors::Vector{Tensor})
```

With this change, we will be able to call `TensorNetwork` with a `Vector` of `Tensor`s  without having to explicitly do `Tensor[tensor1, tensor2, ...]`.

Reply:
I think it would be better if we just do `convert(Vector{Tensor}, tensors)` inside `TensorNetwork`'s inner constructor.